test/integration/test_kraken_order_book_tracker.py

```diff
@@ -0,0 +1,119 @@
+                return
+            await asyncio.sleep(1)
+
+    async def run_parallel_async(self, *tasks, timeout=None):
+        future: asyncio.Future = safe_ensure_future(safe_gather(*tasks))
+        timer = 0
+        while not future.done():
+            if timeout and timer > timeout:
+                raise Exception("Time out running parallel async task in tests.")
+            timer += 1
+            now = time.time()
+            next_iteration = now // 1.0 + 1
+            await asyncio.sleep(1.0)
+        return future.result()
+
+    def run_parallel(self, *tasks):
+        return self.ev_loop.run_until_complete(self.run_parallel_async(*tasks))
+
+    def setUp(self):
+        self.event_logger = EventLogger()
+        for event_tag in self.events:
+            for trading_pair, order_book in self.order_book_tracker.order_books.items():
+                order_book.add_listener(event_tag, self.event_logger)
+
+    def test_order_book_trade_event_emission(self):
+        """
+        Test if order book tracker is able to retrieve order book trade message from exchange and
+        emit order book trade events after correctly parsing the trade messages
+        """
+        self.run_parallel(self.event_logger.wait_for(OrderBookTradeEvent))
+        for ob_trade_event in self.event_logger.event_log:
+            self.assertTrue(type(ob_trade_event) == OrderBookTradeEvent)
+            self.assertTrue(ob_trade_event.symbol in self.trading_pairs)
+            self.assertTrue(type(ob_trade_event.timestamp) in [float, int])
+            self.assertTrue(type(ob_trade_event.amount) == float)
+            self.assertTrue(type(ob_trade_event.price) == float)
+            self.assertTrue(type(ob_trade_event.type) == TradeType)
+            self.assertTrue(math.ceil(math.log10(ob_trade_event.timestamp)) == 10)
+            self.assertTrue(ob_trade_event.amount > 0)
+            self.assertTrue(ob_trade_event.price > 0)
+
+    def test_tracker_integrity(self):
+        # Wait 5 seconds to process some diffs.
+        self.ev_loop.run_until_complete(asyncio.sleep(5.0))
+        order_books: Dict[str, OrderBook] = self.order_book_tracker.order_books
+        xbtusd_book: OrderBook = order_books["XBT/USD"]
+        xrpusdt_book: OrderBook = order_books["XRP/USD"]
+        # print(xbtusd_book.snapshot)
+        # print(xrpusdt_book.snapshot)
+        self.assertGreaterEqual(xbtusd_book.get_price_for_volume(True, 10).result_price,
+                                xbtusd_book.get_price(True))
+        self.assertLessEqual(xbtusd_book.get_price_for_volume(False, 10).result_price,
+                             xbtusd_book.get_price(False))
+        self.assertGreaterEqual(xrpusdt_book.get_price_for_volume(True, 10000).result_price,
+                                xrpusdt_book.get_price(True))
+        self.assertLessEqual(xrpusdt_book.get_price_for_volume(False, 10000).result_price,
+                             xrpusdt_book.get_price(False))
+
+
```

Comment:
It seems like the `OrderBookTracker` is not applying the diffs.

![image](https://user-images.githubusercontent.com/6755922/70468436-f2801000-1a7b-11ea-9773-d33162392cc2.png)